### PR TITLE
Added raw JSON document support to Data

### PR DESCRIPTION
**Pull request by russelarms**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Today String.class is not supported in the Java SDK and triggers various bugs (even documents that are impossible to delete).

We will use the String type as special case to pass raw JSON payloads so that it can interoperate with Xamarin and other platforms.

The String must be JSON and will be stored as a document object (not as a string).

## Related PRs or issues

List related PRs and other issues.

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java [hunk 1]
```diff
@@ -187,6 +190,58 @@ public static void setTokenExchangeUrl(String tokenExchangeUrl) {
         return create(partition, documentId, document, documentType, new WriteOptions());
     }
 
+    /**
+     * Create a document from a JSON String.
+     *
+     * @param jsonDocument must be serializable JSON value.
```
Comment (MatkovIvan) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java]:
```suggestion
     * @param jsonDocument must be serialized JSON value.
```

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java [hunk 2]
```diff
@@ -187,6 +190,58 @@ public static void setTokenExchangeUrl(String tokenExchangeUrl) {
...
+     */
+    static AppCenterFuture<DocumentWrapper<JSONObject>> create(String partition, String documentId, String jsonDocument) {
+        JSONObject jsonObject = convertToJSONObject(jsonDocument);
+        if (jsonObject == null){
```
Comment (MatkovIvan) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java]: Please format and apply IDE's optimizations and cleanups

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java [hunk 3]
```diff
@@ -187,6 +190,58 @@ public static void setTokenExchangeUrl(String tokenExchangeUrl) {
         return create(partition, documentId, document, documentType, new WriteOptions());
     }
 
+    /**
+     * Create a document from a JSON String.
+     *
+     * @param jsonDocument must be serializable JSON value.
+     */
+    static AppCenterFuture<DocumentWrapper<JSONObject>> create(String partition, String documentId, String jsonDocument) {
```
Comment (MatkovIvan) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java]: No need to overload without `WriteOptions`

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java [hunk 4]
```diff
@@ -233,6 +288,23 @@ public static void setDataStoreRemoteOperationListener(DataStoreEventListener li
         getInstance().mEventListener = listener;
     }
 
+    @Nullable
+    private static JSONObject convertToJSONObject(String document){
+        JSONObject jsonObject = null;
+        try {
+            jsonObject = new JSONObject(document);
+        } catch (JSONException e) {
+            AppCenterLog.debug(LOG_TAG, "Can not convert to JSONObject");
```
Comment (MatkovIvan) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java]: Please keep the exception message

```suggestion
            AppCenterLog.debug(LOG_TAG, "Can not convert to JSONObject", e);
```

**Comment by guperrot** [comment]
We will use JsonElement.class from Xamarin instead.

**Pull request closed without merging**